lib/learn_test/strategies/jest.rb

```diff
@@ -0,0 +1,91 @@
+module LearnTest
+  module Strategies
+    class Jest < LearnTest::Strategy
+      def service_endpoint
+        '/e/flatiron_jest'
+      end
+
+      def detect
+        package = File.exists?('package.json') ? Oj.load(File.read('package.json'), symbol_keys: true) : nil
+        return false if !package
+
+        if package[:scripts] && package[:scripts][:test]
+          return true if package[:scripts][:test].include?('jest')
+        end
+
+        if package[:devDependencies] && package[:devDependencies][:jest]
+          return true if (package[:devDependencies][:jest].length > 0)
+        end
+
+        if package[:dependencies] && package[:dependencies][:jest]
+          return true if (package[:dependencies][:jest].length > 0)
+        end
```

Comment:
I'm tempted to have this logic become helper methods available to all strategies because this is very similar to the methods in the mocha strategy. Maybe prefixed with `node` just because non-JS strategies aren't gonna be looking for package.

thinking:
```ruby
def js_package
  # memoized return of package.json hash
end

def has_js_dependency?(dep)
  [:dependencies, :devDependencies].any? { |key| js_package[key] && js_package[key][dep] }
end

has_js_dependency?(:jest)
```

then update both mocha and jest strategy to use those helpers